### PR TITLE
call-signature option of typedef: check arrow function

### LINKED ISSUE #1015: Typedef `call-signature` is not working on arrow functions
See the following example:

```javascript
export class TsLint {
    function helloWorld() {
        return 'Hello World';
    };

    arrowHelloWorld = () => {
        return helloWorld();
    };
}
```

In `tslint.json` file I have the following rule configured:
```javascript
"typedef": [
      true,
      "call-signature",
      "parameter",
      "arrow-parameter",
      "property-declaration",
      "variable-declaration",
      "member-variable-declaration"
    ],
```

When running tslint I get an error only for the `helloWorld` function and not for the arrow function `arrowHelloWorld ` also:
```console
[08:59:38] Starting 'tslint'...
[08:59:38] [gulp-tslint] error (typedef) tslintIssue.ts[2, 25]: expected call-signature: 'helloWorld' to have a typedef
[08:59:38] 'tslint' errored after 175 ms
[08:59:38] Error in plugin 'gulp-tslint'
Message:
    Failed to lint: tslintIssue.ts[2, 25]: expected call-signature: 'helloWorld' to have a typedef.
```

Is this an issue in tslint?

### PR DESCRIPTION
**Pull request by YuichiNukiyama**
fix #1015

### CODE REVIEW

**Comment by JKillian** [comment]
@YuichiNukiyama I'm wondering if we should make this a separate option named "arrow-call-signature"? Also see https://github.com/palantir/tslint/pull/1176 which is somewhat related

**Comment by YuichiNukiyama** [comment]
`arrow-call-signature` option is reasnable. I follow your advise.

File: src/rules/typedefRule.ts [hunk 1]
```diff
@@ -38,6 +38,12 @@ class TypedefWalker extends Lint.RuleWalker {
         super.visitFunctionExpression(node);
     }
 
+    public visitArrowFunction(node: ts.FunctionLikeDeclaration) {
+        const location = (node.parameters != null) ? node.parameters.end : null;
+        this.checkTypeAnnotation("arrow-call-signature", location, node.type, node.name);
+        this.walkChildren(node);
```
Comment (JKillian) [inline src/rules/typedefRule.ts]: Probably better to use `super.visitArrowFunction(node)` here

File: test/rules/typedef/all/test.ts.lint [hunk 2]
```diff
@@ -40,3 +40,21 @@ var NoTypesFn = function (
...
+                   ~          [expected member-variable-declaration: 'arrowHelloWorld' to have a typedef]
+                       ~      [expected arrow-call-signature to have a typedef]
+        return this.helloWorld();
+    };
+}
+
+setTimeout(() => {}, 1000);
+            ~                [expected arrow-call-signature to have a typedef]
+setTimeout(function() {}, 1000);
+                    ~            [expected call-signature to have a typedef]
```
Comment (JKillian) [inline test/rules/typedef/all/test.ts.lint]: Can we add a test or two with usages like this?
`someFunc(n => n+1);` or maybe `someFunc(n => {})`.
Want to make sure it works okay when there's just one argument without parenthesis

File: test/rules/typedef/all/test.ts.lint [hunk 3]
```diff
@@ -40,3 +40,25 @@ var NoTypesFn = function (
...
+
+    arrowHelloWorld = () => {
+                   ~          [expected member-variable-declaration: 'arrowHelloWorld' to have a typedef]
+                       ~      [expected arrow-call-signature to have a typedef]
+        return this.helloWorld();
+    };
+}
+
+setTimeout(() => {}, 1000);
+setTimeout(function() {}, 1000);
```
Comment (YuichiNukiyama) [inline test/rules/typedef/all/test.ts.lint]: `arrow-parameter` option don't warn this line. Is this `arrow-parameter` option bug?

**Comment by YuichiNukiyama** [comment]
@JKillian Thanks for your review.

File: test/rules/typedef/all/test.ts.lint (hunk 3 above)
Comment (JKillian) [inline test/rules/typedef/all/test.ts.lint]: Do you mean `arrow-call-signature`? There aren't any parameters so I wouldn't expect `arrow-parameter` to do anything

Comment (YuichiNukiyama) [inline test/rules/typedef/all/test.ts.lint]: I was mistaken. Don't care :smile:

**Comment by JKillian** [comment]
Thanks @YuichiNukiyama!

**Pull request merged**